tdrs-frontend/src/index.js

```diff
@@ -24,8 +24,28 @@ axios.defaults.xsrfCookieName = 'csrftoken'
 axios.defaults.xsrfHeaderName = 'X-CSRFToken'
 axios.defaults.withCredentials = true
 
+function devLogin(devEmail) {
+  const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
+  axios
+    .post(`${BACKEND_URL}/login/cypress`, {
+      username: devEmail,
+      token: 'local-cypress-token',
+    })
+    .then(function (response) {
+      console.log(response)
+    })
+    .catch(function (error) {
+      console.log(error)
+    })
+  store.dispatch({ type: 'SET_AUTH', payload: { devEmail } })
```

Comment:
i think the reason the profile page comes up blank is because the reducer expects a `user` object in the payload, something like

```js
{
    first_name: 'string',
    last_name: 'string',
    stt: 12,
    roles: [
        'Developer'
    ]
}
```

the `fetchAuth` action function sets the auth state with the result of calling the `/auth_check` endpoint, i think this initial dispatch gets quickly overridden by that auth check, allowing the other pages to work. the profile page, however, immediately looks for the user's `roles` in state, which aren't set by this dispatch, and throws an exception. It's likely the cypress tests would have the same issue if they navigated to the profile page.

there's a couple ways to possibly address this 
- the bug in the profile page could be fixed, allowing the page to load and maybe replace the state with a `fetchAuth` call thereby covering up the initial dispatch payload
- change the dispatch payload to include the roles. this could be setting them blank and letting a `fetchAuth` take over like the above point, or by modifying what the `/cypress/login` endpoint returns to include the roles and put the entire response in state with this dispatch

Reply:
Put some more time into this. A one char fix neatly resolves the bug, allowing initialization to complete. I am then seeing the expected auth with the Developer profile loaded quickly afterwards. Thanks for the pointer!